components/class-m-chart-admin.php

```diff
@@ -237,23 +232,38 @@ public function current_screen( $screen ) {
 			wp_enqueue_script(
 				'm-chart-admin',
 				$this->plugin_url . '/components/js/m-chart-admin.js',
-				array( 'highcharts', 'jquery' ),
+				array( 'jquery' ),
 				m_chart()->version
 			);
 
-			$settings = m_chart()->get_settings();
-			$post_id  = isset( $_GET['post'] ) ? absint( $_GET['post'] ) : '';
+			// Only load this if we are on an appropriate post page
+			$library = isset( $_GET['post'] ) ? m_chart()->get_post_meta( $_GET['post'], 'library' ) : m_chart()->get_library();
+
+			if ( 'post' == $screen->base && 'chartjs' == $library ) {
+				wp_enqueue_script(
+					'm-chart-chartjs-admin',
+					$this->plugin_url . '/components/js/m-chart-chartjs-admin.js',
+					array( 'm-chart-admin', 'chartjs', 'jquery' ),
+					$this->version
+				);
+			}
+
+			$post_id = isset( $_GET['post'] ) ? absint( $_GET['post'] ) : '';
```

Comment:
Maybe move this line above line 239 and use `$post_id` instead of `$_POST['post']` in the assignment of `$library`?